internal/webhook/clusterdeployment_webhook.go

```diff
@@ -296,3 +304,29 @@ func isCredMatchTemplate(cred *kcmv1.Credential, template *kcmv1.ClusterTemplate
 
 	return nil
 }
+
+func ValidateCrossNamespaceRefs(ctx context.Context, namespace string, serviceSpec *kcmv1.ServiceSpec) (errs error) {
+	l := ctrl.LoggerFrom(ctx)
+
+	l.Info(fmt.Sprintf("Validating that the references in .spec.serviceSpec.TemplateRefs do not refer to any resource outside the %s namespace", namespace))
+	for _, ref := range serviceSpec.TemplateResourceRefs {
+		if ref.Resource.Namespace != namespace {
```

Comment:
Can `ref.Resource.Namespace` be omitted? If so, will the template resource be fetched from the same namespace as the cluster deployment or from the default namespace? I just want to confirm that this specific scenario is validated correctly.
Same question for `valuesFrom` validation.

Reply:
Here is the description for `ref.Resource.Name` (i.e., `TemplateResourceRef`): https://projectsveltos.github.io/sveltos/template/intro_template/#templateresourcerefs-namespace-and-name

And here is the description for `ValuesFrom`: https://github.com/projectsveltos/addon-controller/blob/7c7c8f81f135e4773bd53e828eda294315bfda65/api/v1beta1/spec.go#L123-L129

Reply:
Thanks, @wahabmk. If the `ref.Resource.Namespace` is left blank, the resource will correctly be fetched from the same namespace as the cluster, which is expected. However, in the current implementation, this results in a validation error:
```
<Kind> <name> is in namespace "", cannot refer to a resource in a namespace other than <clusterDeploymentNamespace> in .spec.serviceSpec.templateResourceRefs
```
We should handle empty namespace case. Or am I misunderstood something?

Reply:
You're right. I added a check for empty namespace, should be good now. Thanks!